modules/core/src/main/java/org/apache/synapse/unittest/Assertor.java

```diff
@@ -0,0 +1,459 @@
+    }
+
+    /**
+     * Method of assertionNotNull for Sequence test cases.
+     *
+     * @param assertNotNull array of assertNotNull
+     * @param msgCtxt       message context
+     * @return Pair<Boolean, String> which has status of assertNotNull and message if any error occurred
+     */
+    private static Pair<Boolean, String> startAssertNotNullsForSequence(List<AssertNotNull> assertNotNull,
+                                                                        MessageContext msgCtxt) {
+        logger.info("\n");
+        logger.info("---------------------Assert Not Null---------------------\n");
+        boolean isAssertNotNullFailed = false;
+        String messageOfAssertNotNull = null;
+
+        for (AssertNotNull assertItem : assertNotNull) {
+
+            if (!isAssertNotNullFailed) {
+
+                String actual = assertItem.getActual();
+                String message = assertItem.getMessage();
+                boolean isAssertNull;
+                String[] actualType = actual.split(":");
+                String mediatedResult;
+
+                switch (actualType[0]) {
+
+                    case INPUT_PROPERTY_BODY:
+                        mediatedResult = Trimmer.trimStrings(
+                                msgCtxt.getEnvelope().getBody().getFirstElement().toString());
+                        isAssertNull = mediatedResult.isEmpty();
+
+                        break;
+
+                    case INPUT_PROPERTY_CONTEXT:
+                        mediatedResult = Trimmer.trimStrings(msgCtxt.getProperty(actualType[1]).toString());
+                        isAssertNull = mediatedResult.isEmpty();
+
+                        break;
+
+                    case INPUT_PROPERTY_AXIS2:
+                        Axis2MessageContext axis2smc = (Axis2MessageContext) msgCtxt;
+                        org.apache.axis2.context.MessageContext axis2MessageCtx =
+                                axis2smc.getAxis2MessageContext();
+
+                        mediatedResult = Trimmer.trimStrings(axis2MessageCtx.getProperty(actualType[1]).toString());
+                        isAssertNull = mediatedResult.isEmpty();
+
+                        break;
+
+                    case INPUT_PROPERTY_TRANSPORT:
+                        Axis2MessageContext axis2smcTra = (Axis2MessageContext) msgCtxt;
+                        org.apache.axis2.context.MessageContext axis2MessageCtxTransport =
+                                axis2smcTra.getAxis2MessageContext();
+                        Object headers = axis2MessageCtxTransport.getProperty(
+                                org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+
+                        @SuppressWarnings("unchecked")
+                        Map<String, Object> headersMap = (Map) headers;
+
+                        mediatedResult = Trimmer.trimStrings(headersMap.get(actualType[1]).toString());
+                        isAssertNull = mediatedResult.isEmpty();
```

Comment:
Code duplication. Please check in `INPUT_PROPERTY_CONTEXT`, `INPUT_PROPERTY_AXIS2`, `INPUT_PROPERTY_TRANSPORT`

```
 mediatedResult = Trimmer.trimStrings(headersMap.get(actualType[1]).toString());
 isAssertNull = mediatedResult.isEmpty();
```

Reply:
Done